### PR TITLE
[REVIEW] Remove duplicate definitions from docs

### LINKED ISSUE #4528: [DOC] Detail gather's documentation is wrong
## Report incorrect documentation

**Location of incorrect documentation**
https://github.com/rapidsai/cudf/blob/0242d70fe2f4870a5e28e27361d4fe2215fbcd7c/cpp/include/cudf/detail/gather.cuh#L416-L444

**Describe the problems or issues found in the documentation**
Documentation mentions arguments that the function no longer takes. The invalid arguments are `check_bounds` and `allow_negative_indices`

**Steps taken to verify documentation is incorrect**
Read the code that immediately follows

**Suggested fix for documentation**
Remove arguments from doc and the related description and `@throws` or add them in implementation

### PR DESCRIPTION
**Pull request by devavret**
Uses `@copydoc` to remove duplicate definitions from detail headers.
Depends on #5049 
Also Closes #4528

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

File: cpp/include/cudf/unary.hpp [hunk 1]
```diff
@@ -72,7 +72,7 @@ std::unique_ptr<cudf::column> unary_operation(
  * @param input A `column_view` as input
  * @param mr Optional, The resource to use for all allocations
  *
- * @returns std::unique_ptr<cudf::column> A non-nulalble column of `BOOL8` elements with `true`
+ * @returns A non-nulalble column of `BOOL8` elements with `true`
```
Comment (davidwendt) [inline cpp/include/cudf/unary.hpp:75]:
```suggestion
 * @returns A non-nullable column of `BOOL8` elements with `true`
```

File: cpp/include/cudf/unary.hpp [hunk 2]
```diff
@@ -86,7 +86,7 @@ std::unique_ptr<cudf::column> is_null(
  * @param input A `column_view` as input
  * @param mr Optional, The resource to use for all allocations
  *
- * @returns std::unique_ptr<cudf::column> A non-nulalble column of `BOOL8` elements with `false`
+ * @returns A non-nulalble column of `BOOL8` elements with `false`
```
Comment (davidwendt) [inline cpp/include/cudf/unary.hpp:89]:
```suggestion
 * @returns A non-nullable column of `BOOL8` elements with `false`
```

File: cpp/include/cudf/unary.hpp [hunk 3]
```diff
@@ -118,8 +118,7 @@ std::unique_ptr<column> cast(column_view const& input,
  * @param input A column of floating-point elements
  * @param mr Optional, The resource to use for allocating the device memory in the returned column.
  *
- * @returns unique_ptr<column> A non-nulalble column of `BOOL8` elements with `true`
- * representing `NAN` values
+ * @returns A non-nulalble column of `BOOL8` elements with `true` representing `NAN` values
```
Comment (davidwendt) [inline cpp/include/cudf/unary.hpp:121]:
```suggestion
 * @returns A non-nullable column of `BOOL8` elements with `true` representing `NAN` values
```

File: cpp/include/cudf/unary.hpp [hunk 4]
```diff
@@ -135,8 +134,7 @@ std::unique_ptr<column> is_nan(
  * @param input A column of floating-point elements
  * @param mr Optional, The resource to use for allocating the device memory in the returned column.
  *
- * @returns unique_ptr<column> A non-nulalble column of `BOOL8` elements with `false`
- * representing `NAN` values
+ * @returns A non-nulalble column of `BOOL8` elements with `false` representing `NAN` values
```
Comment (davidwendt) [inline cpp/include/cudf/unary.hpp:137]:
```suggestion
 * @returns A non-nullable column of `BOOL8` elements with `false` representing `NAN` values
```

File: cpp/include/cudf/binaryop.hpp [hunk 5]
```diff
@@ -33,40 +33,39 @@ namespace experimental {
...
+  BITWISE_XOR,           ///< operator ^
+  LOGICAL_AND,           ///< operator &&
+  LOGICAL_OR,            ///< operator ||
+  COALESCE,              ///< operator x,y  x is null ? y : x
+  GENERIC_BINARY,        ///< generic binary operator to be generated with input
+                         ///< ptx code
+  SHIFT_LEFT,            ///< operator <<
+  SHIFT_RIGHT,           ///< operator >>
+  SHIFT_RIGHT_UNSIGNED,  ///< operator >>> (from Java)
+  ///< Logical right shift. Casts to an unsigned value before shifing.
```
Comment (davidwendt) [inline cpp/include/cudf/binaryop.hpp:63]: Can this be indented like line 59 above?

**Comment by harrism** [comment]
rerun tests

**Pull request merged**